File: PAT/BLevel/1008/1008.cpp
// 1008 数组元素循环右移问题
// auguralpha
// 2019-02-11
#include <cstdio>
const int maxn = 110;
int arr[maxn];

int main() {
    int n, m;
    scanf("%d%d", &n, &m);
    for(int i = 0; i < n; ++i) {
        scanf("%d", &arr[i]);
    }
    m = m % n;
    for(int i = 0; i < n; ++i) {
        printf("%d%s", arr[(n-m+i)%n], ((i - n + 1) ? " " : "\n"));
    }
    return 0;
}

File: PAT/BLevel/1008/1008.go
// 1008 数组元素循环右移问题
// auguralpha
// 2020-03-25

package main

import "fmt"

func main() {
	var n, m int
	var arr []int
	_, _ = fmt.Scanf("%d %d ", &n, &m)
	m = m % n
	for i := 0; i < n; i ++ {
		var a int
		_, _ = fmt.Scanf("%d", &a)
		arr = append(arr, a)
	}

	for i := 0; i < len(arr); i ++ {
		if i - n + 1 == 0 {
			fmt.Printf("%d\n", arr[(n - m + i) % n])
		} else {
			fmt.Printf("%d ", arr[(n - m + i) % n])
		}

	}
}
